strawberryfields/apps/clique.py

```diff
@@ -169,8 +190,14 @@ def grow(clique: list, graph: nx.Graph, node_select: str = "uniform") -> list:
     if not is_clique(graph.subgraph(clique)):
         raise ValueError("Input subgraph is not a clique")
 
+    if isinstance(node_select, (list, np.ndarray)):
+        if len(node_select) != graph.number_of_nodes():
+            raise ValueError("Number of node weights must match number of nodes")
+        w = {n: node_select[i] for i, n in enumerate(graph.nodes)}

```

Comment:
Why exactly is `w` a set and not a list or array?

Reply:
`w` is a dictionary mapping from graph nodes to their corresponding weights.
E.g.
```python
graph = nx.complete_graph(8)
node_select = [1, 2, 3, 1, 8, 4, 2, 6]
w = {n: node_select[i] for i, n in enumerate(graph.nodes)}
print(w)
# {0: 1, 1: 2, 2: 3, 3: 1, 4: 8, 5: 4, 6: 2, 7: 6}
```
This can matter if the graph is composed of nodes that are not just `list(range(dim))`